doajtest/unit/test_bll_site_sitemap.py

```diff
@@ -0,0 +1,185 @@
+        self.store_tmp_impl = app.config["STORE_TMP_IMPL"]
+        self.store_impl = app.config["STORE_IMPL"]
+
+        self.cache = models.cache.Cache
+        models.cache.Cache = ModelCacheMockFactory.in_memory()
+        models.Cache = models.cache.Cache
+
+        self.toc_changefreq = app.config["TOC_CHANGEFREQ"]
+        app.config["TOC_CHANGEFREQ"] = "daily"
+
+        self.static_pages = app.config["STATIC_PAGES"]
+        app.config["STATIC_PAGES"] = [
+            ("/about", "monthly"),
+            ("supporters", "weekly")
+        ]
+
+    def tearDown(self):
+        app.config["STORE_TMP_IMPL"] = self.store_tmp_impl
+        app.config["STORE_IMPL"] = self.store_impl
+        if os.path.exists("test_store"):
+            shutil.rmtree("test_store")
+        self.localStore.delete_container(self.container_id)
+        self.tmpStore.delete_container(self.container_id)
+
+        models.cache.Cache = self.cache
+        models.Cache = self.cache
+
+        app.config["TOC_CHANGEFREQ"] = self.toc_changefreq
+        app.config["STATIC_PAGES"] = self.static_pages
+
+        super(TestBLLSitemap, self).tearDown()
+
+    @parameterized.expand(load_cases)
+    def test_sitemap(self, name, kwargs):
+
+        prune_arg = kwargs.get("prune")
+        tmp_write_arg = kwargs.get("tmp_write")
+        main_write_arg = kwargs.get("main_write")
+        raises_arg = kwargs.get("raises")
+
+        ###############################################
+        ## set up
+
+        raises = EXCEPTIONS.get(raises_arg)
+        prune = True if prune_arg == "True" else False if prune_arg == "False" else None
+
+        if tmp_write_arg == "fail":
+            app.config["STORE_TMP_IMPL"] = StoreMockFactory.no_writes_classpath()
+
+        if main_write_arg == "fail":
+            app.config["STORE_IMPL"] = StoreMockFactory.no_writes_classpath()
+
+        journals = []
+        for s in JournalFixtureFactory.make_many_journal_sources(count=10, in_doaj=True):
+            j = models.Journal(**s)
+            j.save()
+            journals.append(j)
+        models.Journal.blockall([(j.id, j.last_updated) for j in journals])
+
+        expectations = [(j.bibjson().get_preferred_issn(), j.last_updated) for j in journals]
+
+        if prune:
+            self.localStore.store(self.container_id, "sitemap__doaj_20180101_0000_utf8.xml", source_stream=StringIO("test1"))
+            self.localStore.store(self.container_id, "sitemap__doaj_20180601_0000_utf8.xml", source_stream=StringIO("test2"))
+            self.localStore.store(self.container_id, "sitemap__doaj_20190101_0000_utf8.xml", source_stream=StringIO("test3"))
+
+
+        ###########################################################
+        # Execution
+
+        if raises is not None:
+            with self.assertRaises(raises):
+                self.svc.sitemap(prune)
+
+                tempFiles = self.tmpStore.list(self.container_id)
+                assert len(tempFiles) == 0
```

Comment:
Could we please start adding the messages to test assertion:
`assert len(tempFiles) == 0, "expected 0, received {}".format(len(tempFiles))`

This makes debugging tests much more effective.